app/src/main/java/com/github/swent/echo/data/room/entity/UserProfileRoom.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.swent.echo.data.room.entity
+
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.Index
+import androidx.room.Junction
+import androidx.room.PrimaryKey
+import androidx.room.Relation
+import com.github.swent.echo.data.model.UserProfile
+import com.github.swent.echo.data.model.toSectionEPFL
+import com.github.swent.echo.data.model.toSemesterEPFL
+
+@Entity
+data class UserProfileRoom(
+    @PrimaryKey val userId: String,
+    val name: String,
+    val semester: String?,
+    val section: String?,
```

Comment:
I already started adding association committee memberships as well as association following as relations between `UserProfile` and `Association` over at #157. So this might be something you already want to take a look at depending on which PR gets merged first.

Reply:
OK, thanks for the info, I'll take a look!
By the way, thank you so much for taking the time to review all this!!!